source/grid/grid_tools.cc

```diff
@@ -491,6 +482,60 @@ namespace GridTools
         return false;
       }
     };
+
+
+    /**
+     * get_coarse_mesh_description() needs to store face data for
+     * dim>1, but we can not have this code in the function, as this
+     * requires and instantiation of CellData<dim-1> or constexpr
+     * if. Use a class instead for now.
+     */
+    template <int dim>
+    class FaceDataHelper
+    {
+    public:
+      template <class FaceIteratorType>
+      void
+      add(const FaceIteratorType &face)
+      {
+        CellData<dim - 1> face_cell_data;
+        for (unsigned int vertex_n = 0;
+             vertex_n < GeometryInfo<dim>::vertices_per_face;
+             ++vertex_n)
+          face_cell_data.vertices[vertex_n] = face->vertex_index(vertex_n);
+        face_cell_data.boundary_id = face->boundary_id();
+        face_cell_data.manifold_id = face->manifold_id();
+
+        face_data.insert(face_cell_data);
+      }
+
+      void
+      copy(SubCellData &subcell_data)
```

Comment:
That's a poor name for a function since it's not clear what is being copied from/to where. Can't you just call this function `get()` and let it return the `SubCellData` object you're creating here?